tests/MongoDB.Bson.Tests/BsonExtensionMethodsNominalTypeTests.cs

```diff
@@ -0,0 +1,171 @@
+﻿/* Copyright 2010-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using FluentAssertions;
+using MongoDB.Bson.Serialization;
+using System;
+using Xunit;
+
+namespace MongoDB.Bson.Tests
+{
+    public class BsonExtensionMethodsNominalTypeTests
+    {
+        [Fact]
+        public void ToBson_generic_should_default_args_nominal_type_to_TNominalType()
+        {
+            var c = new C { X = 1 };
+
+            var bytes = c.ToBson();
+
+            var document = BsonSerializer.Deserialize<BsonDocument>(bytes);
+            document.Should().Be("{ X : 1 }");
```

Comment:
Before the fix the incorrect result was:

    { _t : "C", X : 1 }

Reply:
when I removed all changes except tests, this test has passed. I missed something?

Reply:
Line 89 of old version of BsonExtensionMethods.cs.

This bug was not everywhere, but I've moved the defaulting to the argument checking section of the method.

Reply:
yes, I just wanted to highlight that the first comment above is not suitable directly for this test

Reply:
Sorry... the comment was true for most of the cases but I guess I carelessly put it on the wrong case.